_posts/2020-10-23-deconvolution.md

```diff
@@ -0,0 +1,446 @@
+  <line x1="16" y1="6" x2="136" y2="6" />
+  <line x1="18" y1="8" x2="138" y2="8" />
+  <line x1="20" y1="10" x2="140" y2="10" />
+  <line x1="22" y1="12" x2="142" y2="12" />
+  <line x1="24" y1="14" x2="144" y2="14" />
+  <line x1="26" y1="16" x2="146" y2="16" />
+  <line x1="28" y1="18" x2="148" y2="18" />
+  <line x1="30" y1="20" x2="150" y2="20" />
+  <line x1="32" y1="22" x2="152" y2="22" />
+  <line x1="34" y1="24" x2="154" y2="24" />
+  <line x1="36" y1="26" x2="156" y2="26" />
+  <line x1="38" y1="28" x2="158" y2="28" />
+  <line x1="41" y1="31" x2="161" y2="31" />
+  <line x1="42" y1="32" x2="162" y2="32" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="10" y1="0" x2="42" y2="32" style="stroke-width:2" />
+  <line x1="130" y1="0" x2="162" y2="32" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="10.000000,0.000000 130.000000,0.000000 162.743566,32.743566 42.743566,32.743566" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Horizontal lines -->
+  <line x1="42" y1="32" x2="162" y2="32" style="stroke-width:2" />
+  <line x1="42" y1="152" x2="162" y2="152" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="42" y1="32" x2="42" y2="152" style="stroke-width:2" />
+  <line x1="162" y1="32" x2="162" y2="152" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="42.743566,32.743566 162.743566,32.743566 162.743566,152.743566 42.743566,152.743566" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Text -->
+  <text x="102.743566" y="172.743566" font-size="1.0rem" font-weight="100" text-anchor="middle" >2048</text>
+  <text x="182.743566" y="92.743566" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,182.743566,92.743566)">2048</text>
+  <text x="16.371783" y="156.371783" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,16.371783,156.371783)">950</text>
+</svg>
+</td>
+</tr>
+</table>
+
+
+What we now have is a Dask array composed of 16 CuPy blocks of data.  Notice
+how Dask provides nice typing information in the SVG output.  When we moved
+from NumPy to CuPy, the block diagram above displays `Type: cupy.ndarray` --
+this is a nice sanity check.
+
+The last piece we need before running the deconvolution is the PSF which should
+also be loaded onto the GPU:
+
+```python
+avg_psf = skimage.io.imread("/public/NVMICROSCOPY/AVG_PSF.tif")
+c_avg_psf = cp.asarray(avg_psf)
+```
+
+Lastly, we map the `deconvolve` function onto each block in our Dask array:
+
+```
+out = da.map_blocks(deconvolve,
+                    c_imgs,
+                    c_psf,
+                    iterations=20,
+                    dtype=cp.float32)
+```
+
+With Dask and multiple GPUs, we measured deconvolution of an 8GB image in ~10
+seconds! But this is just the first step towards accelerated image science.
+
+<a href="/images/deconvolve.png">
+    <img src="/images/deconvolve.png" width="100%"></a>
+
+IMAGE DESCRIPTION
```

Comment:
I am especially interested to see this! A higher resolution image might help too.

Reply:
Added a description. Also having included a new image. Not sure whether it is better. Please let us know :)